no-littering.el

```diff
@@ -77,36 +77,64 @@ This variable has to be set before `no-littering' is loaded.")
   "The directory where packages place their persistent data files.
 This variable has to be set before `no-littering' is loaded.")
 
-(cl-flet ((etc (file) (expand-file-name (convert-standard-filename file)
-                                        no-littering-etc-directory))
-          (var (file) (expand-file-name (convert-standard-filename file)
-                                        no-littering-var-directory)))
-  (with-no-warnings ; many of these variables haven't been defined yet
-
-;;; Built-in packages
-
-    (setq auto-save-list-file-prefix       (var "saves-"))
-    (setq backup-directory-alist           (list (cons "." (var "backups/"))))
-    (setq bookmark-default-file            (var "bookmarks.el"))
-    (setq desktop-path                     (list (var "desktop/")))
-    (setq desktop-base-file-name           (var "default.el"))
-    (setq desktop-base-lock-name           (var "default.lock"))
-    (setq eshell-directory-name            (var "eshell/"))
-    (setq ido-save-directory-list-file     (var "ido.last"))
-    (setq nsm-settings-file                (var "network-security.data"))
-    (setq org-id-locations-file            (var "org/org-id-locations"))
-    (setq org-registry-file                (var "org/org-registry.el"))
-    (setq recentf-save-file                (var "recentf.el"))
-    (setq save-place-file                  (var "saveplace.el"))
-    (setq savehist-file                    (var "savehist.el"))
-    (setq tramp-persistency-file-name      (var "tramp.el"))
-    (setq trash-directory                  (var "trash/"))
-    (setq url-configuration-directory      (var "url/"))
-
-;;; Third-party packages
-
-    (setq smex-save-file                   (var "smex-items"))
-    ))
+(defun no-littering-make-path (base-dir package file)
+  "Returns path BASE-DIR/[PACKAGE/]FILE, creating the necessary
```

Comment:
The `[PACKAGE/]` part doesn't seem correct. This always uses just `FILE`, but `FILE` may have the form `PACKAGE-DIRECTORY/FILE-NAME`. Generally I think the doc-strings still need some work.